index.js

```diff
@@ -59,3 +60,68 @@ export default function elementReady(selector, {
 
 	return Object.assign(promise, {stop: () => stop()});
 }
+
+export function observeReadyElements(selector, {
+	target = document,
+	stopOnDomReady = true,
+	waitForChildren = true,
+	timeout = Number.POSITIVE_INFINITY,
+} = {}) {
+	return {
+		[Symbol.asyncIterator]() {
+			const {next, complete, onCleanup, iterator} = createDeferredAsyncIterator();
+
+			function handleMutations(mutations) {
+				for (const {addedNodes} of mutations) {
```

Comment:
I think we could reduce this to one loop by running `.flatMap` on this first.

Reply:
`.flatMap()` does not work on NodeLists unless we use the spread operator which would result in a performance drop. This is the cleanest way.